tests/acl/test_mod_wrapper.cpp

```diff
@@ -0,0 +1,178 @@
+#include <string_view>
+
+#include "test_only/test_framework/redemption_unit_tests.hpp"
+#include "test_only/test_framework/check_img.hpp"
+#include "test_only/front/fake_front.hpp"
+#include "test_only/core/font.hpp"
+
+#include "gdi/screen_info.hpp"
+#include "utils/timebase.hpp"
+#include "utils/colors.hpp"
+#include "keyboard/keymap2.hpp"
+#include "core/client_info.hpp"
+#include "RAIL/client_execute.hpp"
+#include "core/RDP/capabilities/window.hpp"
+#include "configs/config.hpp"
+#include "utils/sugar/zstring_view.hpp"
+//#include "utils/png.hpp"
+
+#include "acl/mod_wrapper.hpp"
+
+#define IMG_TEST_PATH FIXTURES_PATH "/img_ref/acl/"
+
+namespace
+{
+    class TestOSDMessageDisplayFixture
+    {
+    public :
+        TestOSDMessageDisplayFixture() :
+            client_info_(get_client_info()),
+            front_(client_info_.screen_info),
+            palette_(BGRPalette::classic_332()),
+            font_(global_font_deja_vu_14()),
+            gd_(front_.gd()),
+            rail_client_execute_(time_base_,
+                                 gd_,
+                                 front_,
+                                 client_info_.window_list_caps,
+                                 false),
+            mod_wrapper_(time_base_,
+                         palette_,
+                         gd_,
+                         keymap_,
+                         client_info_,
+                         font_,
+                         rail_client_execute_,
+                         ini_)
+        {}
+
+    protected :
+        TimeBase time_base_;
+        ClientInfo client_info_;
+        FakeFront front_;
+        const BGRPalette& palette_;
+        Keymap2 keymap_;
+        const Font& font_;
+        gdi::GraphicApi& gd_;
+        ClientExecute rail_client_execute_;
+        Inifile ini_;
+        ModWrapper mod_wrapper_;
+
+    private :
+        static ClientInfo get_client_info()
+        {
+            ClientInfo client_info;
+
+            client_info.screen_info.bpp = BitsPerPixel{24};
+            client_info.screen_info.width = 800;
+            client_info.screen_info.height = 600;
+
+            return client_info;
+        }
+    };
+
+    using namespace std::string_view_literals;
+}
+
+RED_FIXTURE_TEST_CASE(TestOSDMessageDisplay_NormalUrgency,
+                      TestOSDMessageDisplayFixture)
+{
+    constexpr auto osd_message = "Hello World !"sv;
+    gdi::OsdMsgUrgency omu = gdi::OsdMsgUrgency::NORMAL;
```

Comment:
Why not `constexpr auto` as with `osd_message` ?

Reply:
It's already `constexpr auto` the `osd_message` variable (to force the instruction execution at compile time)